# --- T2-COPYRIGHT-BEGIN ---
# t2/package/*/llvm/ia64-libc.diff
# Copyright (C) 2026 The T2 SDE Project
# SPDX-License-Identifier: GPL-2.0 or patched project license
# --- T2-COPYRIGHT-END ---

diff --git llvm/libc/src/__support/FPUtil/ManipulationFunctions.h llvm/libc/src/__support/FPUtil/ManipulationFunctions.h
index 02dded2f4791..6e3d29de803d 100644
--- llvm/libc/src/__support/FPUtil/ManipulationFunctions.h
+++ llvm/libc/src/__support/FPUtil/ManipulationFunctions.h
@@ -275,9 +275,9 @@ LIBC_INLINE constexpr T nextupdown(T x) {
 } // namespace fputil
 } // namespace LIBC_NAMESPACE_DECL
 
-#ifdef LIBC_TYPES_LONG_DOUBLE_IS_X86_FLOAT80
+#if defined(LIBC_TYPES_LONG_DOUBLE_IS_X86_FLOAT80) && defined(LIBC_TARGET_ARCH_IS_X86)
 #include "x86_64/NextAfterLongDouble.h"
 #include "x86_64/NextUpDownLongDouble.h"
-#endif // LIBC_TYPES_LONG_DOUBLE_IS_X86_FLOAT80
+#endif // LIBC_TYPES_LONG_DOUBLE_IS_X86_FLOAT80 && LIBC_TARGET_ARCH_IS_X86
 
 #endif // LLVM_LIBC_SRC___SUPPORT_FPUTIL_MANIPULATIONFUNCTIONS_H
